File: Daftar Belanja/DaftarBelanja.dart
import 'dart:io';

void main() {
  ShoppingList shoppingList = ShoppingList();
  late String input;

  do {
    print('==============================');
    print('Aplikasi Daftar belanja');
    print('==============================');
    print('1. Tambah barang ke daftar belanja');
    print('2. Lihat daftar belanja');
    print('3. Edit daftar belanja');
    print('4. Hapus daftar belanja');
    print('5. Keluar');
    print('==============================');
    input = stdin.readLineSync() ?? '';

    switch (input) {
      case '1':
        shoppingList.AddShoppingList();
        break;
      case '2':
        shoppingList.SeeShoppingList();
        break;
      case '3':
        shoppingList.EditShoppingList();
        break;
      case '4':
        shoppingList.DeleteShoppingList();
        break;
      case '5':
        print('Terima kasih!');
        break;
      default:
        print('Input tidak valid');
    }
  } while (input != '5');
}

class ShoppingList {
  int index = -1;
  late List<String> NamaBarang = [];
  late List<int> JumlahBarang = [];

  void AddShoppingList() {
    stdout.write('Masukkan Nama Barang : ');
    String NamaBarang = stdin.readLineSync() ?? '';
    this.NamaBarang.add(NamaBarang);

    stdout.write('Masukkan Jumlah Barang : ');
    int JumlahBarang = int.parse(stdin.readLineSync() ?? '0');
    this.JumlahBarang.add(JumlahBarang);

    index++;
  }

  void SeeShoppingList() {
    for (int i = 0; i < NamaBarang.length; i++) {
      print('${i + 1}. ${NamaBarang[i]} (${JumlahBarang[i]})');
    }
  }

  void EditShoppingList() {
    stdout.write('Masukkan ID Barang yang ingin diubah : ');
    int IDBarang = int.parse(stdin.readLineSync() ?? '0');

    if (IDBarang > index || IDBarang < 0) {
      print('ID tidak ditemukan');
      return;
    }

    stdout.write('Masukkan Nama Barang baru : ');
    String NamaBarangBaru = stdin.readLineSync() ?? '';

    stdout.write('Masukkan Jumlah Barang baru : ');
    int JumlahBarangBaru = int.parse(stdin.readLineSync() ?? '0');

    NamaBarang[IDBarang] = NamaBarangBaru;
    JumlahBarang[IDBarang] = JumlahBarangBaru;
  }

  void DeleteShoppingList() {
    print('Masukkan ID Barang yang ingin dihapus : ');
    int IDBarang = int.parse(stdin.readLineSync() ?? '0');

    if (IDBarang > index || IDBarang < 0) {
      print('ID tidak ditemukan');
      return;
    }

    NamaBarang.removeAt(IDBarang);
    JumlahBarang.removeAt(IDBarang);
  }
}

File: Daftar Belanja/DaftarBelanja.c
#include <stdio.h>
#include <stdlib.h>

struct ShoppingList {
    int index;
    char NamaBarang[100][100];
    int JumlahBarang[100];
};

void AddShoppingList(struct ShoppingList *shoppingList) {
    printf("Masukkan Nama Barang : ");
    scanf("%s", shoppingList->NamaBarang[shoppingList->index]);
    
    printf("Masukkan Jumlah Barang : ");
    scanf("%d", &shoppingList->JumlahBarang[shoppingList->index]);
    
    shoppingList->index++;
}

void SeeShoppingList(struct ShoppingList *shoppingList) {
    for (int i = 0; i < shoppingList->index; i++) {
        printf("%d. %s (%d)\n", i + 1, shoppingList->NamaBarang[i], shoppingList->JumlahBarang[i]);
    }
}

void EditShoppingList(struct ShoppingList *shoppingList) {
    int IDBarang;
    printf("Masukkan ID Barang yang ingin diubah : ");
    scanf("%d", &IDBarang);

    if (IDBarang >= shoppingList->index || IDBarang < 0) {
        printf("ID tidak ditemukan\n");
        return;
    }

    printf("Masukkan Nama Barang baru : ");
    scanf("%s", shoppingList->NamaBarang[IDBarang]);

    printf("Masukkan Jumlah Barang baru : ");
    scanf("%d", &shoppingList->JumlahBarang[IDBarang]);
}

void DeleteShoppingList(struct ShoppingList *shoppingList) {
    int IDBarang;
    printf("Masukkan ID Barang yang ingin dihapus : ");
    scanf("%d", &IDBarang);

    if (IDBarang >= shoppingList->index || IDBarang < 0) {
        printf("ID tidak ditemukan\n");
        return;
    }

    for (int i = IDBarang; i < shoppingList->index - 1; i++) {
        strcpy(shoppingList->NamaBarang[i], shoppingList->NamaBarang[i + 1]);
        shoppingList->JumlahBarang[i] = shoppingList->JumlahBarang[i + 1];
    }

    shoppingList->index--;
}

int main() {
    struct ShoppingList shoppingList;
    shoppingList.index = 0;

    char input[10];

    do {
        printf("==============================\n");
        printf("Aplikasi Daftar belanja\n");
        printf("==============================\n");
        printf("1. Tambah barang ke daftar belanja\n");
        printf("2. Lihat daftar belanja\n");
        printf("3. Edit daftar belanja\n");
        printf("4. Hapus daftar belanja\n");
        printf("5. Keluar\n");
        printf("==============================\n");
        scanf("%s", input);

        switch (input[0]) {
            case '1':
                AddShoppingList(&shoppingList);
                break;
            case '2':
                SeeShoppingList(&shoppingList);
                break;
            case '3':
                EditShoppingList(&shoppingList);
                break;
            case '4':
                DeleteShoppingList(&shoppingList);
                break;
            case '5':
                printf("Terima kasih!\n");
                break;
            default:
                printf("Input tidak valid\n");
        }
    } while (input[0] != '5');

    return 0;
}
